igc_lib.py

```diff
@@ -71,7 +74,148 @@ def _rawtime_float_to_hms(timef):
 
     return hms((time/3600), (time%3600)/60, time%60)
 
-
+class Turnpoint:
+    """ single turnpoint in a task.
+    
+    Attributes: 
+        lat: latitude as a float in degrees DD.DD
+        lon: longitude as a float in degrees DD.DD
+        radius: radius of cylinder or line in km
+        sort: type of turnpoint; start_exit, start_enter, cylinder, ESS, goal_cylinder, goal_line
+        
+    """
+    def __init__(self, lat, lon, radius, sort):
+        self.lat = lat
+        self.lon = lon
+        self.radius = radius
+        self.sort = sort
+        
+
+    def in_radius(self, fix):
+        """Computes great circle distance in kilometers to a GNSSFix.
+        returns true if the fix is within the radius"""
+        lat1, lon1, lat2, lon2 = map(math.radians, [self.lat, self.lon, fix.lat, fix.lon])
+        
+        if (EARTH_RADIUS_KM * _sphere_distance(lat1, lon1, lat2, lon2)) < self.radius:
+            return True
+        else:
+            return False
+        
+        
+class Task:
+    
+    
+    @staticmethod
+    def create_from_lkt_file(filename):
+        """ Creates Task from LK8000 task file, which is in xml format.
+            LK8000 does not have ESS or task finish time.
```

Comment:
End of Speed Section. (have expanded). In paragliding this is when the clock stops which is usually a km or 2 before the goal. A safety measure to stop pilots flying full speed to the goal, which is a landing field. i.e. prevent them flying full speed close to the ground. (first pilot to ESS wins as long as they then arrive at goal afterwards)